packages/browser/jest-dom.d.ts

```diff
@@ -260,8 +260,7 @@ declare namespace matchers {
      * @see
      * [testing-library/jest-dom#tohaveclass](https://github.com/testing-library/jest-dom#tohaveclass)
      */
-    toHaveClass(classNames: string, options?: {exact: boolean}): R
-    toHaveClass(...classNames: Array<string | RegExp>): R
+    toHaveClass(...classNames: (string | RegExp)[] | [string, options?: {exact: boolean}]): R
```

Comment:
```suggestion
    toHaveClass(...classNamesAndOption:
    | [string | RegExp, ...(string | RegExp)[]]
    | [string | RegExp, ...(string | RegExp)[], { exact: boolean }]): R
```

I believe the following would permit the correct type, correct me if I'm wrong.

```ts
type Matcher = string | RegExp;
type Option = {exact: boolean};
type Parameters = [Matcher, ...Matcher[]] | [Matcher, ...Matcher[], Option]

...
type toHaveClass(...classNamesAndOption: Parameters): R
```

Below is the exact snippet I have in VSCode and it types correctly.

```ts
type Matcher = string | RegExp;
type Option = {exact: boolean};
type Parameters = [Matcher, ...Matcher[]] | [Matcher, ...Matcher[], Option]


type ToHaveClass = (...classNamesAndOption: Parameters) => boolean;

const toHaveClass: ToHaveClass = (..._args) => {
  return false;
}

toHaveClass("x")
toHaveClass("x", "y", "z");
toHaveClass("x", "y", "z", {exact: true});
toHaveClass("x", /y/, "z", {exact: false});
toHaveClass(/z/, {exact: false});
```

Reply:
Thanks for the suggestion :+1:
We might not want to change the typing from testing-library https://github.com/testing-library/jest-dom/blob/918b6fbcde10d4409ee8f05c6e4eecbe96a72b7a/types/matchers.d.ts#L262, so I think we stick with `[string, options?: {exact: boolean}]` for now, even though they probably supports `{ exact: boolean }` at the last position of any arguments length https://github.com/testing-library/jest-dom/blob/918b6fbcde10d4409ee8f05c6e4eecbe96a72b7a/src/to-have-class.js#L3.